snorkeling_helper/database_helper/database_modules.py

```diff
@@ -0,0 +1,342 @@
+                                if section == "SUPPL"
+                                else section
+                            )
+
+                        # Fix those pesky_citations
+                        section_text = clear_citations("(", ")", section_text)
+                        section_text = clear_citations("[", "]", section_text)
+                        section_text = re.sub(r"\"", "", section_text)
+
+                        doc = nlp(section_text, disable=["ner"])
+                        doc_sentences = list(doc.sents)
+
+                        for position, sentence in enumerate(doc_sentences):
+                            (word, pos_tag, lemma, dep, char_offset) = list(
+                                zip(
+                                    *[
+                                        (
+                                            repr(tok.text),
+                                            repr(tok.tag_),
+                                            repr(tok.lemma_.lower()),
+                                            repr(tok.dep_),
+                                            str(tok.idx + offset),
+                                        )
+                                        for tok in sentence
+                                    ]
+                                )
+                            )
+
+                            writer.writerow(
+                                {
+                                    "document_id": document_id,
+                                    "section": section.lower(),
+                                    "position": position
+                                    + section_counter[
+                                        section
+                                    ],  # sections are broken by paragraphs and titles
+                                    "text": sentence.text,
+                                    "word": "|".join(word),
+                                    "pos_tag": "|".join(pos_tag),
+                                    "lemma": "|".join(lemma),
+                                    "dep": "|".join(dep),
+                                    "char_offset": "|".join(
+                                        char_offset
+                                    ),  # act like title and abstract are one string
+                                }
+                            )
+
+                        # Update the section counter based on sections
+                        section_counter.update({section: len(doc_sentences)})
+
+                    del section_counter
+                    del nlp
+
+            data_queue.task_done()
+
+            # Remove subelements from document
+            # Thereby freeing memory
+            clear_document(document)
+            del document
+
+        except Exception as e:
+            print(f"This was an exception {e}")
+            # Do nothing but tell the queue that the task is done
+            data_queue.task_done()
```

Comment:
Do you need to use `data_queue.task_done()` here if you're calling it right after the try/except block anyway?

Reply:
Yeah I do need this line there because I want to tell the queue that the document had an exception but was "parsed anyways". Other wise if a document has an exception it task done will never be called for it. The last task done call below is for the None element I added onto the queue at the end.